src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -883,6 +884,14 @@ private async Task<ConnectBlocksResult> ConnectBlockAsync(ChainedHeaderBlock blo
                     PeersToBan = badPeers
                 };
 
+                if (validationContext.RejectUntil != null)
+                {
+                    failureResult.BanDurationSeconds = (int)(validationContext.RejectUntil.Value.ToUnixTimestamp() - this.dateTimeProvider.GetAdjustedTimeAsUnixTimestamp());
+
+                    if (failureResult.BanDurationSeconds < 1)
```

Comment:
@noescape00 can you explain the change for us please?

Reply:
Basically this is a chech against a race condition to ensure we don't set `BanDurationSeconds` to 0 or negative value. 

The change overall is to ban peers for time till their blocks become valid. That way we will remove blocks from memory and CHT and when ban expires and we connect to those peers we will start downloading discarded chain again.

Reply:
Shouldn't this chagne go in the `PosFutureDriftRule`? we already calculate reject until at that point, seems to me this is a complementry change to that logic there.

I agree here you can just check 
```
if (failureResult.BanDurationSeconds < 1) 
          failureResult.BanDurationSeconds = 1; 
```

Reply:
@noescape00 ?